### DISCUSSION TITLE
How to convert the Ego-Facebook dataset in SNAPDatasets to a single graph?

### DISCUSSION THREAD

**Discussion by scottshufe** [pyg-team/pytorch_geometric#5606] [Q&A]
Hi, everyone. 

I try to load ego-facebook dataset in SNAPDatasets and I find that it consists of 10 graphs.
I have seen some research works used this dataset for node classification task, and my question is how to convert this dataset to a single graph and how to obtain the node labels for classification?

Thanks in advance.

**Comment by rusty1s** [comment]
Can you point me to the papers that make use of this dataset? The official page specifies that this dataset contains 10 graphs, see [here](https://snap.stanford.edu/data/ego-Facebook.html). It might be possible though that papers just fuse these graphs together into one single graph though.

**Reply by scottshufe** [reply]
This technical report - [Scaling Attributed Network Embedding to Massive Graphs](https://arxiv.org/pdf/2009.00826.pdf) used the facebook as a single graph.
And I find [torch_geometric.datasets.AttributedGraphDataset](https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/datasets/attributed_graph_dataset.html#AttributedGraphDataset) contains this facebook dataset. Maybe the authors just fuse the ego-graphs to form the single graph, and the authors say that it treats each ego-graph as the node label in the paper.
